### DISCUSSION TITLE
Throw exception but also send acknowledgement?

### DISCUSSION THREAD

**Discussion by drawnwren** [miguelgrinberg/python-socketio#1389] [General]
Is there another way to send a callback acknowledgement to a client from within `python-socketio` without returning a value? I can't seem to find anything in the docs. 

When we encounter an exception in our application, I would like to send a callback acknowledgement to the client, but also raise the exception after the acknowledgement. I can't see a way to do that with only the return value callback from the server docs.

**Comment by miguelgrinberg** [comment]
You can catch the exception, log it without raising it, then return your acknowledgement.
